cookbook/import_export/product-export.rst

```diff
@@ -0,0 +1,200 @@
+Understand the CSV Product Export
+=================================
+
+We'll discuss here how the csv product export works.
+
+It's a good start to understand the overall architecture and how to re-use or replace some parts.
+
+.. note::
+
+  The import part has been widely re-worked in 1.4, please note that the export part remains the same than in 1.3 (all classes and services are in the BaseConnectorBundle).
+
+Definition of the Job
+---------------------
+
+The product export is defined in ``src\Pim\Bundle\BaseConnectorBundle\Resources\config\batch_jobs.yml``.
+
+.. code-block:: yaml
+
+    connector:
+        name: Akeneo CSV Connector
+        jobs:
+            csv_product_export:
+                title: pim_base_connector.jobs.csv_product_export.title
+                type:  export
+                steps:
+                    export:
+                        title:     pim_base_connector.jobs.csv_product_export.export.title
+                        services:
+                            reader:    pim_base_connector.reader.doctrine.product
+                            processor: pim_base_connector.processor.product_to_flat_array
+                            writer:    pim_base_connector.writer.file.csv_product
+                        parameters:
+                            batch_size: 10
+
+With the ``type`` parameter, we can see that this job is an export.
+
+We can also count a single step named ``export``.
+
+Product Export Step
+-------------------
+
+The purpose of this step is to read products from database, to transform product objects to array and write lines in a csv file.
+
+This step is a default step, an ``Akeneo\Bundle\BatchBundle\Step\ItemStep`` is instanciated and injected.
+
+.. code-block:: yaml
+
+    [...]
+    export:
+        title:         pim_base_connector.jobs.csv_product_export.export.title
+        services:
+            reader:    pim_base_connector.reader.doctrine.product
+            processor: pim_base_connector.processor.product_to_flat_array
+            writer:    pim_base_connector.writer.file.csv_product
+        parameters:
+            batch_size: 10
+    [...]
+
+An ``ItemStep`` always contains 3 elements, a ``Akeneo\Bundle\BatchBundle\Item\ItemReaderInterface``, a ``Akeneo\Bundle\BatchBundle\Item\ItemProcessorInterface`` and a ``Akeneo\Bundle\BatchBundle\Item\ItemWriterInterface``.
+
+We provide here specific implementations for these elements, the services are declared with aliases ``pim_base_connector.reader.doctrine.product``, ``pim_base_connector.processor.product_to_flat_array``, ``pim_base_connector.writer.file.csv_product``.
+
+Product Reader
```

Comment:
Can you add a note, just to introduce how we load storage storage driver. You can just add something like "To know more about how we load different storage driver you can see https://github.com/akeneo/pim-community-dev/blob/master/src/Pim/Bundle/CatalogBundle/DependencyInjection/PimCatalogExtension.php#L56"